BuildScripts/mvComplexFunctions.sql

```diff
@@ -917,6 +917,7 @@ BEGIN
 
             tLastType               := tDmlType;
             iArraySeq               := 1;
+            uRowIDArray             := '{}';
```

Comment:
This section of code is batching the processing of log table entries into groups with the same change type. So if there are 8 `UPDATE` changes, then 8 `DELETE` changes, it'll do the updates all in one batch, then the deletes all in one batch. 

The problem was when you get e.g. 8 `UPDATE` changes followed by say 4 `DELETE` changes, it'll batch the 8 updates, process them, and then batch the 4 deletes but _use the same array as before_, so the indexes that haven't been overwritten still have the last 4 updates in them. That results in those 4 getting deleted from the materialized view.

Reply:
Hi, thanks for your update. That's great news you're looking at what we've done so far and potentially getting some use out of it. Yes, we've seen a number of issues with combo of changes which have been fixed including this one - but due to our client needs we've had to put some customized changes to meet their requirements so we've done this against our internal branches for now but this are all live and working now off a separate branch. But i will get confirmation if it's ok to include all of this in this master branch as the custom one's will only get used if your materialized views are named the same as ours. Thanks

Reply:
Great, thank you! Could you point me to the branch that has other fixes on it, in case we ought to patch our version with some of the other issues you've found?